Перевёрстка

Liknara opened the pull request:
Вёрстка по шаблону
Оно живое.. ЖИВОЕ!

Liknara committed:
Сверстал на уикенде 3 блока.. ну не совсем..
Хэдер, стрип и пониже него
Пытаюсь сделать так чтобы страница была всегда по центру, добавляя себе проблем, видимо
По фану добавил фон, шоб с краёв не было пустоты и границы страниц лучше видеть
Добавил нужные шрифты, картинки.. ну не совсем те, но того же размера, просто люблю пошутить, надеюсь не выганят
Надеюсь мастеркласс прояснит ряд моменетов

Liknara committed:
Сверстал на уикенде 3 блока.. ну не совсем..
Хэдер, стрип и пониже него
Пытаюсь сделать так чтобы страница была всегда по центру, добавляя себе проблем, видимо
По фану добавил фон, шоб с краёв не было пустоты и границы страниц лучше видеть
Добавил нужные шрифты, картинки.. ну не совсем те, но того же размера, просто люблю пошутить, надеюсь не выганят
Надеюсь мастеркласс прояснит ряд моменетов

Liknara committed:
Переделываю вёрстку с нуля
Сделал хэдер
Начал линки
Всеми силами избегаю дивы, вроде даже выходит
(На самом деле это 2-я версия переделки)

Liknara committed:
Переделываю вёрстку с нуля
Сделал хэдер
Начал линки
Всеми силами избегаю дивы, вроде даже выходит
(На самом деле это 2-я версия переделки)

Liknara committed:
Разместил страницу по центру экрана
Переделал косяки с {margin: -x} и другие
Сделеал более логичные классы и, следовательно, больше не больше не ссылаюсь по типу [.класс селектор]
Допёр что выделение слов красным можно сделать классом для спана и не засорять css
Есть проблема с отступом, который не убрать если не прописать команду *{margin: 0; padding: 0;}. Оно написано в body, но не работает.

Liknara committed:
Подредактировал то что было, заменил таблицу на спаны - неубирающийся отступ отступил, надо разобраться как работают списки
Набросал следущие пару секций в html

Liknara committed:
Отредактировл beige-stripe - смотриться лучше и сделанно проще
Сверстал секцию block-1

Liknara committed:
Сверстал секцию block-1
Заметил что экран не скролится, тк изначально спозиционировал страницу через fixed, пока поменял на absolut. Надо посмотреть что с этим можно сделать.. или ничего не делать. Хз.

Liknara committed:
Сверстал секцию gray-stripe
Заметил что экран не скролится, тк изначально спозиционировал страницу через fixed, пока поменял на absolut. Надо посмотреть что с этим можно сделать.. или ничего не делать. Хз.

Liknara committed:
Merge remote-tracking branch 'origin/Перевёрстка' into Перевёрстка

Liknara committed:
Изменил вставку картинок - теперь через переменные
Отредактировал полоску линков под хэдером
Добавил секцию "block-2" и сопутствующую "block-2__stripe"

Liknara committed:
Избавился от "position absolute";
Что-то подправил в "block-2";
Сверстал секцию "video", но гит хаб сказал "иди на хер со своими 100мб", так что кина не будет;
Сверстал секцию "articles";
Сверстал секцию "footer";
Верстаю секцию "community".

Liknara committed:
Доделал секцию "community".

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,671 @@
+.header__spine {
+    display: block;
+    margin: 10rem 6.5rem 0 2rem;
+    padding: 0 0 0 0;
```
лучше написать `Padding: 0`

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,671 @@
+    font-size: 4.875rem;
+    font-weight: 700;
+    line-height: 5.85rem;
+    margin: 6.4rem 0 0 0;
```
лучше написать `margin-top: 6.4rem`, когда применяется margin или padding лишь на одну сторону, то правильнее написать margin/padding-"выбранная сторона": "значение"

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,671 @@
+}
+
+.header__paragraph {
+    font-size: 1rem;
```
Размеры шрифтов пишутся в пикселях, а не ремах

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,671 @@
+}
+
+.header__find-button {
+    font-size: 1rem;
```
Шрифт в пикселях

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,671 @@
+    background: var(--gray);
+    max-width: 11rem;
+    box-sizing: border-box;
+    padding: 0.75rem 1.3rem 0.75rem 3rem;
```
0 в дробных числах с нулем в целой части мы не пишем. 0.75 -> .75

Liknara replied:
Можешь уточнить почему и обязательно ли так? Мне удобнее читать в формате 0.0.

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,671 @@
+    --back-color: #711010;
+    --dark-blue: #232d5a;
+
+    --header-logo: url("/img/logo.png");
+    --header-background: url("/img/header-background.png");
+    --header--menu-button: url("/img/header-menu-button.png");
+    --beige-stripe-1-stripe: url("/img/stripe.png");
+    --beige-stripe-1-spotyfy: url("/img/spotify.png");
+    --beige-stripe-1-microsoft: url("/img/microsoft.png");
+    --beige-stripe-1-dropbox: url("/img/dropbox.png");
+    --beige-stripe-1-behance: url("/img/behance.png");
+    --block-1-background: url("/img/block-1-background.png");
+    --block-1-list: url("/img/arrows.png");
+    --block-2-image-1: url("/img/block-2-image-1.png");
+    --block-2-image-2: url("/img/block-2-image-2.png");
+    --article-1: url("/img/article-1.png");
+    --article-2: url("/img/article-2.png");
+    --article-3: url("/img/article-3.png");
+    --article-4: url("/img/article-4.png");
+    --team-man: url("/img/team-man.png");
+    --team-nigga: url("/img/team-nigga.png");
+    --team-woman: url("/img/team-woman.png");
+    --team-link-1: url("/img/team-link-1.png");
+    --team-link-2: url("/img/team-link-2.png");
+    --team-link-3: url("/img/team-link-3.png");
+    --patients: url("/img/patients.png");
+    --news-1: url("/img/news-1.png");
+    --news-2: url("/img/news-2.png");
+    --news-3: url("/img/news-3.png");
```
Есть ссылки на изображения, которые названы не верно и из-за этого не отображаются в окончательном варианте верстки (такие как спотифай, майкрософт, дропбокс и беханс)

Liknara replied:
У меня в edge, chrome и встроенном в idea браузерах отображаются spotify, microsoft и прочие картинки. Либо я что-то не так понял в замечании.. Прошу уточнить

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,671 @@
+}
+
+.beige-stripe__links-2 {
+    content: var(--beige-stripe-1-spotyfy);
```
не отображается изображение

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,671 @@
+}
+
+.beige-stripe__links-3 {
+    content: var(--beige-stripe-1-microsoft);
```
не отображается изображение

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,671 @@
+}
+
+.beige-stripe__links-4 {
+    content: var(--beige-stripe-1-dropbox);
```
не отображается изображение

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,671 @@
+}
+
+.beige-stripe__links-5 {
+    content: var(--beige-stripe-1-behance);
```
не отображается изображение

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,671 @@
+
+.block-1__list {
+    list-style-image: var(--block-1-list);
+    font-size: 1rem;
```
шрифты в пикселях

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,671 @@
+}
+
+.box-title {
+    font-size: 3rem;
```
шрифты в пикселях

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,671 @@
+.box-paragraph {
+    color: var(--transparent-gray);
+    font-family: var(--inter);
+    font-size: 1rem;
```
шрифты в пикселях

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,671 @@
+
+.column-name {
+    font-family: var(--inter);
+    font-size: 1rem;
```
шрифты в пикселях

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,671 @@
+}
+
+.column-title {
+    font-size: 2.125rem;
```
шрифты в пикселях

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,671 @@
+
+.column-paragraph {
+    font-family: var(--inter);
+    font-size: 1rem;
```
шрифты в пикселях

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,671 @@
+
+.column-button {
+    font-family: var(--inter);
+    font-size: 0.75rem;
```
шрифты в пикселях

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,671 @@
+    margin-left: 3.25rem;
+}
+
+.column-name {
+    font-family: var(--inter);
+    font-size: 1rem;
+    font-weight: 400;
+    line-height: 1.6rem;
+}
+
+.column-title {
+    font-size: 2.125rem;
+    font-weight: 700;
+    line-height: 3.4rem;
+}
+
+.column-paragraph {
+    font-family: var(--inter);
+    font-size: 1rem;
+    font-weight: 400;
+    line-height: 1.6rem;
+}
+
+.column-button {
+    font-family: var(--inter);
+    font-size: 0.75rem;
+    font-weight: 500;
+    line-height: 1.2rem;
+}
```
пишешь одни и те же значение шрифта, не проще ли было вынести отдельным модификатором или в родительский блок общие черты, чтобы не дублировать код.

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,671 @@
+    justify-content: right;
+    max-width: 23.6875rem;
+    max-height: 23.1875rem;
+    margin: 8.75rem 0 0 0;
```
проще написать `margin-top: 8.75rem`

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,671 @@
+
+.beige-stripe__title {
+    margin-top: 5.25rem;
+    font-size: 2.5rem;
```
шрифты в пикселях

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,671 @@
+
+.beige-stripe__paragraph {
+    font-family: var(--inter);
+    font-size: 1rem;
```
ну и так далее короче

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,671 @@
+}
+
+.video__list {
+    list-style: none;
```
правильнее и элегантнее было бы указать здесь, чтобы в списке был маркировка была автоматически "-", чем указывать каждый раз "-" в HTML, это хардкод

Liknara replied:
Подумал, раз оно в фигме шрифтом, то и у меня будет шрифтом)

Yashiandr replied:
ну в фигме иначе не сделать =D

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,671 @@
+}
+
+.article-category {
+    font-size: 0.875rem;
```
ну и 0 не пишем в здесь, и так далее везде

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,671 @@
+.community-subtitle {
+    font-family: var(--inter);
+    font-size: 1rem;
+    margin: 8rem 0 0 0;
```
и так далее

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,671 @@
+
+/*TEAM*/
+
+.community__team {
+}
```
лишнее

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,671 @@
+}
+
+.patients__block {
+    background: var(--patients) no-repeat;
+    padding-left: 37.8725rem;
```
хардкод, проще было разместить контент справа блока или же вставить картинку отдельным блоком, чем двигать на 38 ремов текст

Yashiandr commented on index.html:
```diff
@@ -7,6 +7,273 @@
+            <div class="news__column">
+                <div class="news__column-1">
+                    <h3 class="news__column-title _news-width">The Best Fonts for Modern Design</h3>
+                    <p class="news__column-date">By <a href="http://123.cum" class="news__author">Admin</a> - January 25, 2021</p>
```
текст не того цвета

Yashiandr commented on index.html:
```diff
@@ -7,6 +7,273 @@
+                    saepe dolores sed malorum sapientem, copiosae et melnusquam nam an abhorreant.</p>
+            </div>
+            <ul class="video__list">
+                <li class="video__list-point">– Project strategy</li>
+                <li class="video__list-point">– UX research and conceptualization</li>
+                <li class="video__list-point">– Information architecture (IA)</li>
+                <li class="video__list-point">– User interface design (UI)</li>
+                <li class="video__list-point">– Interaction design (IxD)</li>
```
Лучше сделать отображение маркера списка в css`ке

Yashiandr commented on index.html:
```diff
@@ -7,6 +7,273 @@
+            <p class="header__paragraph">Et ullum iisque conclusionemque eam, mel ad erat cum
+                accumsan noluisse, nostrum accumsan movet salutandi.
+                Fugit expetenda interesset no, probo eloquentiam mei eu. </p>
+            <h2 class="header__find-button"><a href="http://123.cum"> Find Our more </a></h2>
```
не сделана стрелка у кнопки, если не знаешь как, то в телеге в мастерклассах есть ссылка на фигму и гайд на ютуб, как добавить элемент на мастер-класс Ивана, чтобы он разобрал, как делается данный элемент.

Yashiandr reviewed (commented):
В целом неплохо, но куча элементов, которые не отображаются, по типу стрелок, картинок, чуть недоделанное, где-то цвета не те, где-то хардкод, старайся использовать margin в ситуациях с крайней необходимостью и подстраивать контент на их место другими методами. 

Также отсортируй папку с изображениями, чтобы они лежали не в одной куче, а упорядоченно, те что относятся к хедеру в папке с хедером, те, что относятся к одному блоку в папке с названием этого блока и так далее, общие для всего сайта картинки можно уже хранить в общей папке, либо же в специальной созданной для них (например логотипы)

Ну и в дальнейшем необходимо, конечно, добавить анимации на кнопки и другие элементы дизайна.

Liknara committed:
font-size to px
pic to directory

Liknara committed:
font-size to px
pic to directory

Liknara committed:
Merge remote-tracking branch 'origin/Перевёрстка' into Перевёрстка

# Conflicts:
#	css/main.css

Liknara committed:
font-size to px
pic to directory
The button at the top right has been converted into a header
Less hardcode
Made a hover for links in the footer

Liknara committed:
Добавил ховер для кнопок с классом .box-link
Требуется его доработка

Liknara committed:
Доработал ховер для кнопок с классом .box-link
Раскидал картинки по папкам и поправил названия переменных c учётом введения block-0

Liknara committed:
Заменил часть marging и padding flex размещением

Liknara committed:
Изменил расположение links в block-0. Теперь ровнее и без margin-padding. Для адекратного вида пришлось добавить border.
Изменил размещение маркеров списка с list-style на before, чтобы убрать padding
Заменил очередную часть marging и padding flex размещением
Ещё несколько hover-ов и доработка старых

Liknara committed:
Заменил очередную часть marging и padding flex размещением
 Отредактировал сексию коммунити

Liknara committed:
Заменил очередную часть marging и padding flex размещением
Работаю над перевёрсткой footer

Liknara committed:
Переделал footer
